packages/mobile/src/dappsExplorer/DAppsExplorerScreen.tsx

```diff
@@ -0,0 +1,349 @@
+  Image,
+  ScrollView,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View,
+} from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { dappsListApiUrlSelector } from 'src/app/selectors'
+import BackButton from 'src/components/BackButton'
+import BottomSheet from 'src/components/BottomSheet'
+import Dialog from 'src/components/Dialog'
+import CustomHeader from 'src/components/header/CustomHeader'
+import DappsExplorerLogo from 'src/icons/DappsExplorerLogo'
+import Help from 'src/icons/navigator/Help'
+import QuitIcon from 'src/icons/QuitIcon'
+import { TopBarIconButton } from 'src/navigator/TopBarButton'
+import useSelector from 'src/redux/useSelector'
+import { navigateToURI } from 'src/utils/linking'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'DAppExplorerScreen'
+
+interface CategoryWithItems {
+  id: string
+  name: string
+  fontColor: string
+  backgroundColor: string
+  items: DappItem[]
+}
+
+interface DappItem {
+  id: string
+  iconUrl: string
+  name: string
+  description: string
+  dappUrl: string
+}
+
+export function DAppsExplorerScreen() {
+  const { t } = useTranslation()
+  const dappsListUrl = useSelector(dappsListApiUrlSelector)
+  const [categoriesWithItems, setcategoriesWithItems] = useState<CategoryWithItems[]>([])
+  const [isHelpDialogVisible, setHelpDialogVisible] = useState(false)
+  const [isBottomSheetVisible, setBottomSheetVisible] = useState(false)
+  const [dappSelected, setDappSelected] = useState<DappItem>()
+
+  if (!dappsListUrl) {
+    return (
+      <View style={styles.centerContainer} testID="DAppExplorerScreen/error">
+        <Text style={fontStyles.large}> Configuration error. </Text>
+        <Text style={fontStyles.large}> dappsListUrl is not set. </Text>
+      </View>
+    )
+  }
+
+  const { loading } = useAsync(
+    async () => {
+      const response = await fetch(dappsListUrl, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+        },
+      })
+      return response.json()
+    },
+    [],
+    {
+      onSuccess: (result) => {
+        Logger.debug(TAG, `fetch onSuccess: ${JSON.stringify(result)}`)
+        try {
+          const categoriesById: { [id: string]: CategoryWithItems } = {}
+          result.categories.forEach((cat: any) => {
+            categoriesById[cat.id] = {
+              id: cat.id,
+              name: cat.name,
+              fontColor: cat.fontColor,
+              backgroundColor: cat.backgroundColor,
+              items: [],
+            }
+          })
+          result.applications.forEach((app: any) => {
+            categoriesById[app.categoryId].items.push({
+              id: app.id,
+              name: app.name,
+              iconUrl: app.logoUrl,
+              description: app.description,
+              dappUrl: app.url,
+            })
+          })
+          setcategoriesWithItems(Object.values(categoriesById))
```

Comment:
So this has the potential to cause an error (not sure if leak memory or what) because this is an async function that keeps a reference to the `setcategoriesWithItems` callback. If the user goes back or something this `setcategoriesWithItems` callback may not be defined anymore or not sure exactly what can happen. I think we can just return the `Object.values(categoriesById)` and it will be available in the `result` field that `useAsync` returns.

Maybe @jeanregisser can correct me if that's not the real problem or add more context 😅

Reply:
Oh, nice catch! I will change it, so it returns the value instead of calling `setCategoryiesWithItems`

Reply:
Exactly @gnardini 👍